assets/js/modules/analytics/datastore/profiles.js

```diff
@@ -98,12 +100,14 @@ const baseActions = {
 	 * @since 1.8.0
 	 *
 	 * @param {string} propertyID Google Analytics property ID.
+	 * @param {string} profileName The name for a new profile.
 	 * @return {Object} Object with `response` and `error`.
 	 */
-	*createProfile( propertyID ) {
+	*createProfile( propertyID, profileName ) {
```

Comment:
@felixarntz @tofumatt what do you guys think about `profileName` here? Should it be a positional argument or in an object? I [made the case earlier that it should be required](https://github.com/google/site-kit-wp/pull/1680#discussion_r441443813) and I still think it should be, but it just seems odd to have a name as a positional parameter. What if we wanted to support customizing other details of a profile in the future? Obviously we wouldn't want to keep tacking on positional arguments. It's a bit odd here only because I think this is the first action where we've had to make a decision like this. Most other parameters for other actions and selectors are IDs.

Reply:
Yeah, I agree: I think `*createProfile( propertyID, { name } = {} ) {` would be the more appropriate function signature here. Leaves room for customisation, for something that could easily get more arguments. 👍

Reply:
It would be `*createProfile( propertyID, { name } )` since the profile name is required 😄

Reply:
Using an object sounds good to me here, and yes, the object should not have a default then 👍

Reply:
@eugene-manuilov after checking with @tofumatt and @felixarntz, we've decided to change the second argument to `createProfile` to be an object which for now will only have `profileName` instead of the positional argument. See https://github.com/google/site-kit-wp/pull/1680#discussion_r442957195

Basically the signature would change like so:
```diff
- *createProfile( propertyID, profileName ) {
+ *createProfile( propertyID, { profileName } ) {
```
- where the second parameter would be an object that we destructure in the signature. As for docblocks, let's call the parameter `args` since it's required, and then `profileName` would be documented as `args.profileName`.

As for `argsToParams` this would destructure it out as well, and the rest would be the same (it wouldn't pass the whole object through in params).

`controlCallback` and `reducerCallback` would be unchanged since those take the result of `argsToParams`. This change should mostly be about updating the usage and tests.

Reply:
Updated.